pcdscalc/be_lens_calcs.py

```diff
@@ -937,3 +927,156 @@ def find_z_pos(energy, lens_set, spot_size_fwhm, material=None,
+                     ' z_range, beam_size_unfocused not defined.')
+        return
+    focus_before_sample = int(focus_before_sample)
+    if size_vertical is not None:
+        size_calc_1 = np.max(size_horizontal, size_vertical)
+    else:
+        size_calc_1 = size_horizontal
+
+    t_lens_radii = LENS_RADII
+    for rad in exclude:
+        t_lens_radii.remove(rad)
+
+    sets, effrads, sizes, foc_lens = calc_lens_set(
+        energy=energy,
+        size_fwhm=size_calc_1,
+        distance=z_offset,
+        n_max=max_tot_number_of_lenses,
+        max_each=max_each,
+        lens_radii=LENS_RADII,
+    )
+
+    distances = np.asarray([
+            calc_distance_for_size(
+                size_calc_1,
+                list(chain(*list(zip(set, t_lens_radii)))),
+                energy=energy,
+                fwhm_unfocused=beam_size_unfocused,
+            )[focus_before_sample] for set in sets])
+
+    good_sets = np.logical_and(
+        distances > np.min(z_offset + np.asarray(z_range)),
+        distances < np.max(z_offset + np.asarray(z_range)))
+
+    sets = sets[good_sets]
+    size_range_min = np.asarray([
+            calc_beam_fwhm(
+                energy,
+                list(chain(*list(zip(set, t_lens_radii)))),
+                distance=z_offset - min(z_range),
+                fwhm_unfocused=beam_size_unfocused,
+                printsummary=False,
+            ) for set in sets])
+
+    size_range_max = np.asarray([
+        calc_beam_fwhm(
+            energy,
+            list(chain(*list(zip(set, t_lens_radii)))),
+            distance=z_offset - max(z_range),
+            fwhm_unfocused=beam_size_unfocused,
+            printsummary=False,
+        ) for set in sets])
+
+    sizes = sizes[good_sets]
+    effrads = effrads[good_sets]
+    distances = distances[good_sets]
+    foclens = foc_lens[good_sets]
+    transms = np.asarray(
+        [lens_transmission(r=ter, fwhm=beam_size_unfocused, energy=energy)
+            for ter in effrads])
+    Nlenses_s = np.sum(sets, 1)
+
+    resstring = " N   f/m   Min/um   Max/um   T/%  Set  \n"
+    zips = list(
+        zip(sets, size_range_min, size_range_max,
+            effrads, transms, Nlenses_s, foclens)
+    )
+
+    t = "{:2d} {:5.2f} {:8.1f} {:8.1f} {:5.1f}  "
+    for n, z in enumerate(zips):
+        the_set, sizemin, sizemax, eff_rad, transm, n_lenses, foclen = z
+        logger.debug('eff_rad: %s', eff_rad)
+        logger.debug('n_lenses: %s', n_lenses)
+        resstring += t.format(n, foclen, sizemin * 1e6,
+                              sizemax * 1e6, transm * 100)
+        resstring += ", ".join(
+            [
+                "%d x %dum" % (setLensno, t_lens_radii[m] * 1e6)
+                for m, setLensno in enumerate(the_set)
+                if setLensno > 0
+            ]
+        )
+        resstring += "\n"
+    logger.info('\n %s', resstring)
+
+
+def lens_transmission(r, fwhm, n=1, energy=None, id="Be", lens_thicknes=None):
```

Comment:
In the old code we have ID='IF1' - is the ID here the material? that's what is seems like but `IF1` is not a valid material for the current `get_att_len` function
https://github.com/pcdshub/old-hutch-configs/blob/18c7769d66a647ab558ad7ebc34d019fc98fbf48/cxi/blutil/calc.py#L1189

Reply:
Small horror story here: the old hutch python actually used a modified version of `periodictable`. So there very well could have been an added entry named `IF1`.

Reply:
Do you think we'll need to support it?

Reply:
I hope not. I'll quickly go looking for this illicit version to see what it does.

Reply:
Ok, a grep for IF1 is turning up empty

Reply:
Thank you for checking @ZLLentz

Reply:
Actually I just found it here, in the same file you linked: https://github.com/pcdshub/old-hutch-configs/blob/18c7769d66a647ab558ad7ebc34d019fc98fbf48/cxi/blutil/calc.py#L1815

Reply:
ohh noo 😭

Reply:
Ok, I am going to add the aliases here then for the chemical formulas that they have

Reply:
added this in - 1fcd6fb - but I don't like it, should we have those densities and material in a separate file?

Reply:
Keeping them in code is fine for now, in my opinion. Then yaml or JSON down the line.